src/test/java/com/midaswebserver/midasweb/services/UserServiceImpTest.java

```diff
@@ -0,0 +1,441 @@
+package com.midaswebserver.midasweb.services;
+
+import com.midaswebserver.midasweb.models.User.Symbol;
+import com.midaswebserver.midasweb.models.User.User;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.test.util.AssertionErrors.*;
+
+@Transactional
+@SpringBootTest
+public class UserServiceImpTest {
+
+    @Autowired
+    private UserService userService;
+
+    /**
+     * arguably we test the validateUniqueUsername using these userAdd tests
+     */
+    //good test
+    @Test
+    public void goodAddOneUser1() {
+        final User testUser = new User
+                ("CabbageMan", "password", "riblueliany@docomo.ne.jp", "4065556666");
+        assertTrue("Test User Couldn't be added", userService.add(testUser));
+        assertNotNull("User was not found in database", userService.getUserByUsername(testUser.getUsername()));
+    }
+
+    @Test
+    public void goodAddOneUser2() {
+        final User testUser = new User
+                ("MonkeyKing", "PassW0rd", "一个@阿育王。印度", "861065529988");
+        assertTrue("Test User Couldn't be added", userService.add(testUser));
+        assertNotNull("User was not found in database", userService.getUserByUsername(testUser.getUsername()));
+    }
+
+    /**
+     * Checking if more than one user can be added to the database
+     */
+    @Test
+    public void goodUserAddTwoUsers() {
+        final User testUser1 = new User
+                ("MonkeyKing", "PassW0rd", "一个@阿育王。印度", "861065529988");
+        final User testUser2 = new User
+                ("LordCabbage", "CabbageMan", "pleaseImSafe@friendly.edu", "23414443333");
+        assertTrue("TestUser2 wasn't added in add method", userService.add(testUser1));
```

Comment:
assertion comments are not updated, as they state `testUser2` and `testUser3`, but the variable names are `testUser1` and `testUser2`.